Werkfile

```diff
@@ -0,0 +1,25 @@
+let mdbook = which "mdbook"
+let werk-vscode-version = shell "jq --raw-output .version werk-vscode/package.json" 
+
+build "/werk-vscode/werk-{werk-vscode-version}.vsix" {
```

Comment:
What's missing here is running `npm install`.
I'd love to do
```
build "werk-vscode/node_modules" {
    from "werk-vscode/package-lock.json"
    run "npm --prefix werk-vscode install"
}
```
but that will be rebuilt every time because `werk` expects the `node_modules` to be in the `target` folder. I don't think it supports files in the workspace right now?

Reply:
> # Limitations
> *Separate output directory*: It is not possible to put output files next to input files, and files in the output directory are assumed to be generated by werk.

The question is, is this a "not implemented" limitation or "this fundamentally opposes the values of werk" limitation? I think it would be very useful to have but I can understand if you regard this as out of scope.

Reply:
So a build recipe (currently) can't produce a directory, although that would be an amazing feature.

I think the `--prefix` flag to `npm` needs to point into werk's output dir, using a `"<...>"` interpolated variable. I would like to support the syntax `run "npm --prefix <OUT_DIR>/node_modules install"`, but that requires a few features that are still in the pipeline.

Still, I wouldn't be _too_ worried about integrating NPM with werk's outdatedness tracking, as long as `npm` gets the right flags. It would be fine (IMO) for something like this to be a `task` instead of a build recipe, if it can't be supported nicely.

Reply:
> I would like to support the syntax run "npm --prefix <OUT_DIR>/node_modules install", but that requires a few features that are still in the pipeline.

I think `npm --prefix {OUT_DIR}` would work but only if `npm` gets executed from the cwd of `vscode-werk`. I don't think that's possible right now, right?
But a
```
build "werk-vscode/node_modules" {
    from "werk-vscode/package-lock.json"
    cwd "werk-vscode"
    run "npm install --prefix {OUT_DIR}"
}
```
would probably not be too hard to add, and also useful because not every tool has a `-f / --manifest-path / --prefix` flag.

(I also tried `run "sh -c 'cd werk-vscode && npm install'"` but the quotes don't get parsed: `&command_line = Absolute { path: "/usr/bin/sh" } "-c" "\'cd" "werk-vscode" "&&" "npm" "install\'"`)

Reply:
I want `sh -c '...'` to be supported, feel free to create an issue for it.

I'll much around with `npm` a bit and try to figure out what the best solution is here. I bet it's a useful thing that people will be looking for in the docs too.